aips/move-spec-verifier.md

```diff
@@ -0,0 +1,114 @@
+2. Run the Move mutator tool (`mutate`) to generate mutants with the previously specified parameters.
+3. Run the Move Prover tool, passing the generated mutants individually.
+4. Collect the results and generate a report.
+
+The report contains information about the generated mutants as well as the killed ones for each tested source file. All the intermediate results are saved in the configurable (default: `mutants_output`) directory.
+
+## Reference Implementation
+
+TBD
+
+## Risks and Drawbacks
+
+There are the following risks and drawbacks of this proposal:
+- The set of mutation operators may be complete, as it is hard to predict all possible ways of introducing changes in the code. However, the tool is designed to be easily extendable, so it will be possible to add new operators in the future.
+- Verifying the Move specifications can be time-consuming, as it requires running the Move compiler (check if the mutant is valid) and then Move Prover for each mutant. The more mutation operators are enabled and the longer the file is, the more time is required to verify the specifications. There are two ways to mitigate this issue:
+ - The tool can be run in parallel, as each mutant is verified independently.
+ - The tool's parameters can be adjusted to, e.g., run only a subset of mutation operators or downsample the number of mutants.
+
+## Future Potential
+
+Parts of this project, such as the Move Mutator tool, can be used to create mutants for other purposes. For example, it can be used to create a set of mutants and check if the test cases can detect them. So, there is a significant potential to become a test suite verification tool. The main difference would be calling tests instead of the Move Prover.
+
+Thanks to the modular design, the tool can be easily extended to support more mutation operators and categories. It should influence the community to introduce new ways of generating mutants, and therefore, it can increase verification capabilities. That's the potential to produce better specifications and test suites.
+
+## Timeline
+
+### Suggested implementation timeline
+
+The Move Specification Testing tools are expected to be implemented by the end of Q1 2024.
+
+### Suggested developer platform support timeline
+
+Instructions for the developers on how to use the tools and how to extend tools with new operators and categories will be provided by the end of Q1 2024, along with the project implementation.
+
+### Suggested deployment timeline
+
+The Move Specification Testing tools are a part of the Move language ecosystem and are used offline by the developers. Therefore, there is no need to deploy them on the networks.
+
+## Security Considerations
+
+This set of tools does not introduce any changes in the security model. Therefore, it does not have any security implications.
```

Comment:
It is possible for an attacker to use these tools to identify "loose" specifications/tests and use that information to infer problems with code.

Reply:
Added.